File: 프로그래머스/2/169198. 당구 연습/당구 연습.java
class Solution {
    public int[] solution(int m, int n, int startX, int startY, int[][] balls) {
        int[] answer = new int[balls.length];

        for (int i = 0; i < balls.length; i++) {
            int x = balls[i][0];
            int y = balls[i][1];
            int ans = Integer.MAX_VALUE;
            
            int[][] arr = {{startX, startY, x, y}, {m-startX, startY, m-x, y}, {startY, startX, y, x}, {n-startY, startX, n-y, x}};
            for (int[] a : arr) {
                if (a[1] == a[3] && a[0] > a[2]) {
                    continue;
                } else {
                    int temp = (a[0]+a[2]) * (a[0]+a[2]) + (a[1]-a[3]) * (a[1]-a[3]);
                    ans = Math.min(ans, temp);
                }
            }
            
            answer[i] = ans;
        }
        
        return answer;
    }
}

File: 프로그래머스/2/169198. 당구 연습/당구 연습.py
def solution(m, n, startX, startY, balls):
    answer = []
    for i in balls:
        arr = [[startX, startY, i[0], i[1]], [m - startX, startY, m - i[0], i[1]], [startY, startX, i[1], i[0]], [n - startY, startX, n - i[1], i[0]]]
        ans = 0
        for j in arr:
            if j[1] == j[3] and j[0] > j[2]:
                continue
            temp = (j[0]+j[2])**2 + (j[1]-j[3])**2
            if ans == 0:
                ans = temp
            else:
                ans = min(ans, temp)
        answer.append(ans)
    return answer